### PR TITLE
Added missing JDL deployment options

### LINKED ISSUE #14817: Deployment JDL doesn't offer all Kubernetes options
##### **Overview of the issue**

If I run `jhipster k8s` and answer the questions as follows:

```
$ jhipster k8s
microservices
../
select all
No to prometheus
clustered? Yes
Admin password? admin
namespace? default
repo name? gcr.io/wise-name-333
docker push
Enable Istio? No 
service type?
LoadBalancer
dynamic storage? Yes
a specific storage? empty
```

... it'll generate a `.yo-rc.json` that looks like this:

```json
{
  "generator-jhipster": {
    "appsFolders": ["blog", "gateway", "store"],
    "directoryPath": "../",
    "clusteredDbApps": ["store"],
    "serviceDiscoveryType": "eureka",
    "jwtSecretKey": "ODAwNjk0MTViNTllYzAxNjZjNjE5MmE2NjZjMjMwYzdmZTZjNjkyMGQzN2Q3YWEwZWQzZTg2OTdhNzQ3N2QxNmM2ZDJiODFhZmM5N2Y4NmFjODBmOTUzOWRkNjBiMDAwMmFmZQ==",
    "dockerRepositoryName": "gcr.io/jhipster7",
    "dockerPushCommand": "docker push",
    "kubernetesNamespace": "default",
    "kubernetesServiceType": "LoadBalancer",
    "kubernetesUseDynamicStorage": true,
    "kubernetesStorageClassName": "",
    "ingressDomain": "",
    "monitoring": "no",
    "istio": false
  }
}
```

However, it doesn't seem that [our deployment JDL](https://www.jhipster.tech/jdl/deployments) supports the dynamic storage or storage class name options for k8s. The following seems to get close, but doesn't produce the same `.yo-rc.json`.

```
deployment {
  deploymentType kubernetes
  appsFolders [gateway, blog, store]
  clusteredDbApps [store]
  dockerRepositoryName "gcr.io/jhipster7"
}
```

Related: does there need to be a `jwtSecretKey` for OAuth apps?

##### **Motivation for or Use Case**

It'd be nice to create the same configuration with JDL that you can with `jhipster k8s`.

##### **JHipster Version(s)**

7.0.1

### PR DESCRIPTION
**Pull request by MathieuAA**
This PR adds and fixes some JDL deployment options as reported by @mraible on #14817.
This PR is still a draft, so expect some changes.

---

Please make sure the below checklist is followed for Pull Requests.

- [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [x] Tests are added where necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
- [ ] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

Fixes #14817

### CODE REVIEW

**Comment by MathieuAA** [comment]
@mraible this PR is now in a working state, but I'm not comfortable with making "ready for review" just yet.
I'm gonna spend more time reviewing the options because I'm pretty sure there are some things related to deployment options that I haven't updated yet.

**Comment by MathieuAA** [comment]
Some updates: I'm done adding missing deployment options. However, validations are missing so I have to add them.

File: jdl/jhipster/deployment-options.js [hunk 1]
```diff
@@ -16,82 +16,124 @@
...
+    two: 2,
+  },
   storageType: {
     ephemeral: 'ephemeral',
     persistent: 'persistent',
   },
 };
 
-Options.defaults = (deploymentType = Options.deploymentType.dockerCompose) =>
-  _.omitBy(
```
Comment (deepu105) [inline jdl/jhipster/deployment-options.js:74]: I dont remmber why did omitBy here originally, so hope its still works

**Review by deepu105 — APPROVED** [review]
Looks good. But I havent tested it. I'm trusting the tests here

File: jdl/jhipster/deployment-options.js (hunk 1 above)
Comment (MathieuAA) [reply jdl/jhipster/deployment-options.js:74]: The omit here is useful if you have undefined values, and as the previous code generated them, the omit's role was to clean things up.

**Comment by MathieuAA** [comment]
Thanks @DanielFran for the labels! @mraible do you have time to checkout this branch and test it out?

File: jdl/jhipster/deployment-options.js (hunk 1 above)
Comment (MathieuAA) [reply jdl/jhipster/deployment-options.js:74]: And thanks for the review :)

**Comment by mraible** [comment]
I tried this branch with the following JDL:

```
deployment {
  deploymentType kubernetes
  appsFolders [gateway, blog, store]
  clusteredDbApps [store]
  dockerRepositoryName "gcr.io/jhipster7"
  kubernetesNamespace demo
  kubernetesServiceType LoadBalancer
  kubernetesUseDynamicStorage true
}
```

The resulting `.yo-rc.json` has *almost* everything. However, it's missing:

```
"kubernetesStorageClassName": ""
```

If I try to add this to my JDL:

```
deployment {
  deploymentType kubernetes
  appsFolders [gateway, blog, store]
  clusteredDbApps [store]
  dockerRepositoryName "gcr.io/jhipster7"
  kubernetesNamespace demo
  kubernetesServiceType LoadBalancer
  kubernetesUseDynamicStorage true
  kubernetesStorageClassName ""
}
```

I get an error.

```
error: Syntax error message:
	A name is expected, but found: """"
	at line: 9, column: 30
The kubernetesStorageClassName property name must match: /^[A-Za-z]+$/, got "".
	at line: 9, column: 30
Error during import-jdl: Error: A name is expected, but found: """"
	at line: 9, column: 30
The kubernetesStorageClassName property name must match: /^[A-Za-z]+$/, got "".
	at line: 9, column: 30
```

If I use the value that the prompts give me (empty parenthesis instead of blank):

```
deployment {
  deploymentType kubernetes
  appsFolders [gateway, blog, store]
  clusteredDbApps [store]
  dockerRepositoryName "gcr.io/jhipster7"
  kubernetesNamespace demo
  kubernetesServiceType LoadBalancer
  kubernetesUseDynamicStorage true
  kubernetesStorageClassName ()
}
```

It fails too:

```
ERROR! Error during import-jdl: NoViableAltException: Expecting: one of these possible Token sequences:
  1. [BOOLEAN]
  2. [NAME, '.']
  3. [NAME]
  4. ['[']
  5. [INTEGER]
  6. [STRING]
but found: '('
	at line: 9, column: 30
```

**Comment by MathieuAA** [comment]
Okay, the fixed is pushed.
The documentation should be updated for the issue to be truly "closed".

**Comment by mraible** [comment]
LGTM!

**Review by mraible — APPROVED** [review]

**Comment by MathieuAA** [comment]
@mraible if it's okay with you, I wanna create a PR to update the docs before merging it.

**Pull request merged**